pkgs/top-level/all-packages.nix

```diff
@@ -15259,7 +15259,9 @@ with pkgs;
     shards_0_16
     shards;
 
-  shellcheck = callPackage ../development/tools/shellcheck {};
+  shellcheck = callPackage ../development/tools/shellcheck {
+    haskellPackages.ShellCheck = haskellPackages.ShellCheck_0_8_0;
```

Comment:
I would not recommend overriding in this way, as it is brittle at best because it bypasses the `haskellPackages` fixpoint.

Reply:
Can you suggest an alternative?

Reply:
```suggestion
    ShellCheck = haskellPackages.ShellCheck_0_8_0;
```
Then take `ShellCheck` as an input instead of `haskellPackages` in `shellcheck/default.nix`. It is the only attribute used anyways.

Reply:
Updated.

Reply:
Seems alright, the shellcheck expression could probably use a refactor, but that can be done another time (by someone else).

Reply:
What you you have in mind for a refactor? I'll try to find some time in the next few weeks.